app/services/process_feed_service.rb

```diff
@@ -155,7 +153,8 @@ def status_from_xml(entry)
         reply: thread?(entry),
         language: content_language(entry),
         visibility: visibility_scope(entry),
-        conversation: find_or_create_conversation(entry)
+        conversation: find_or_create_conversation(entry),
+        thread: thread?(entry) ? find_status(thread(entry).first) : nil
```

Comment:
Alright, I must have misremembered the conversation from your last PR about thread resolving. Seems fine then.